File: beecrowd/1048 - Aumento de Salário.rs
use std::io;

fn read_float() -> f64 {
    let mut entrada = String::new();

    io::stdin()
        .read_line(&mut entrada)
        .expect("Falha ao ler uma string");

    return entrada.trim()
                    .parse::<f64>()
                    .expect("Falha ao ler como ponto flutuante");
}

fn main() {

    let salario = read_float();
    let mut percentual = i32::default();

    if salario <= 400.0 {
        percentual = 15;
    } else if salario <= 800.0 {
        percentual = 12;
    } else if salario <= 1200.0 {
        percentual = 10;
    } else if salario <= 2000.0 {
        percentual = 7;
    } else {
        percentual = 4;
    }

    let ganho = (percentual as f64) / 100.0 * salario;

    println!("Novo salario: {:.2}", salario + ganho);
    println!("Reajuste ganho: {:.2}", ganho);
    println!("Em percentual: {} %", percentual)
}

File: beecrowd/1048 - Aumento de Salário.go
package main

import "fmt"

func getReajuste(salario float64) int32 {
	if salario <= 400.0 {
		return 15
	} else if salario <= 800.0 {
		return 12
	} else if salario <= 1200.0 {
		return 10
	} else if salario <= 2000.0 {
		return 7
	}
	return 4
}

func main() {
	var salario float64

	fmt.Scan(&salario)

	percentual := getReajuste(salario)
	reajuste := salario * float64(percentual) / 100.0
	novo_salario := salario + reajuste

	fmt.Printf("Novo salario: %.2f\n", novo_salario)
	fmt.Printf("Reajuste ganho: %.2f\n", reajuste)
	fmt.Printf("Em percentual: %d %%\n", percentual)
}
